btclightclient/txout_verification.go

```diff
@@ -0,0 +1,229 @@
+package btclightclient
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/binary"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"io"
+
+	// "fmt"
+
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/wire"
+)
+
+type Hash256Digest [32]byte
+
+type VerifyStatus int
+
+type SPVProof struct {
+	blockHash   *chainhash.Hash
+	txId        *chainhash.Hash
+	txIndex     uint
+	merkleProof []byte
+}
+
+type PMerkleTree struct {
+	numberTransactions uint
+	vBits              []bool
+	vHash              []*chainhash.Hash
+}
+
+
+func readPMerkleTree(r io.Reader, pMerkletree *PMerkleTree, buf []byte) error {
```

Comment:
Why do we provide a buffer as a function argument? Shouldn't we just create it in the function? Also, maybe it's better if we return the PMerkleTree object rather than providing a pointer to it as an argument.